src/support_sphere/lib/logic/cubit/signup_cubit.dart

```diff
@@ -47,16 +74,33 @@ class SignupCubit extends Cubit<SignupState> {
   void setValid() => emit(state.copyWith(isValid: true));
   void setInvalid() => emit(state.copyWith(isValid: false));
 
+  /// Sign up with email and password.
   Future<void> signUpWithEmailAndPassword() async {
+    // If the form is invalid, do nothing
     if (!state.isValid) return;
     emit(state.copyWith(status: FormzSubmissionStatus.inProgress));
     try {
       // TODO: Add coupon code check for signup
-      await _authenticationRepository.signUp(
+      final response = await _authenticationRepository.signUp(
         email: state.email,
         password: state.password,
         signupCode: state.signupCode,
       );
+      User? user = response.user;
+      if (user == null) {
+        // If for some reason the user is null, we should set form status to failure
+        emit(state.copyWith(status: FormzSubmissionStatus.failure));
+        return;
+      }
+
+      // Once we have an authenticated user, we can create the user profile and person
+      await _userRepository.createNewUser(
+        user: user,
+        userName: state.userName,
```

Comment:
Again, I have misgivings on whether or not we NEED usernames in this app. But if we wanted them, they should be unique, right? Where/when would we check for that uniqueness (if and only if we're keeping them in)?

Reply:
We are currently not identifying people by their username but rather their email. We haven't had a discussion whether this should be fully unique or not... typically it is, so would probably need a check in the future like you point out.

Reply:
In my opinion, we should scrap username and keep email. 

We're calling Supabase Auth with emails/passwords, and I have to imagine there's some sort of uniqueness one-user-per-email thing being checked there, or at least it seems that way from docs: https://supabase.com/docs/reference/python/auth-signup

Reply:
since profile is connected to the auth table and it's a 1-1 we still have a uniqueness here, though not exactly a unique username. I doubt people want their email fully exposed to everyone so probably would be good to get people by their username, so I suppose that's a good way to use it. If there are duplicate of usernames, there might be 2 different person. And if those 2 people have the same exact names, then it get's complicated, but probably still shows 2 entities if we retrieve by email. 

I would agree to just remove username, but we need some way to show something for people to refer to each other. One way to mitigate duplicates that I was thinking is if we generate the username based on their names rather than allowing people to to make their own usernames.